### PR TITLE
Pull pom.xml changes off 1.0.0 branch

### PR DESCRIPTION
**Pull request by keefertaylor**
A bunch of changes got made on the 1.0.0 branch to accommodate publishing to Maven Central. 

Pull them back to this branch in order to cut a new release with Neil's fix. 

A few notes:
- Documentation updates where required in order to make the JavaDoc plugin happy. 
- AFAICT, the protocol buffer compiler does not generate JavaDoc and the JavaDoc plugin only allows excludes by packages. Because of this, a large bit of code is excluded. I added a TODO and task in Asana to track and fix this.
- Updated CI config to drop `--remote` which leaves the submodule at the pinned state, rather than pulling from head on master.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,12 @@
 
     <groupId>io.xpring</groupId>
     <artifactId>xpring4j</artifactId>
-    <version>1.0-SNAPSHOT</version>
+    <version>1.0.0</version>
```
Comment (sappenin) [inline pom.xml]: The `master` branch should never have a release version in it. Instead, checkout a tag for `1.0.0` and release from there. The `master` branch should _always_ be a `SNAPSHOT`.

File: src/main/java/io/xpring/javascript/JavaScriptSigner.java [hunk 2]
```diff
@@ -29,8 +29,12 @@ public JavaScriptSigner() throws JavaScriptLoaderException {
 
     /**
      * Sign the given transaction with the given wallet.
+     *
+     * @throws JavaScriptLoaderException An exception if the javascript could not be loaded.
```
Comment (sappenin) [inline src/main/java/io/xpring/javascript/JavaScriptSigner.java]: Minor nit, but generally this should come last in the Javadoc. For future, consider using checkstyle here -- see https://github.com/hyperledger/quilt for details.

Comment (keefertaylor) [reply src/main/java/io/xpring/javascript/JavaScriptSigner.java]: Linting is a tech debt cleanup task that I'm tracking in Asana :)  I've blocked a day a week as sheerly tech debt clean up days so this is a thing that will get burned down soon.

Comment (keefertaylor) [reply src/main/java/io/xpring/javascript/JavaScriptSigner.java]: Added throws

**Pull request merged**